pkg/operator/terminationobserver/storage.go

```diff
@@ -0,0 +1,147 @@
+package terminationobserver
+
+import (
+	"errors"
+	"sort"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// The store interface implementation keeps the in-memory state for each API server.
+// The state consists of current creationTimestamp and previous creationTimestamp as well as termination
+// events recorded by termination observer.
+type Store interface {
```

Comment:
why public?

Reply:
why an interface at all?

Reply:
because I'm used to wrap "storage" into interface (so you can mock them in unit tests), I can drop the interface, I just think it is an anti-pattern in go and it does not really simplify the code.